DynamicProgramming/EggDropping.java

```diff
@@ -20,24 +21,25 @@ private static int minTrials(int n, int m) {
         // j trials for only 1 egg
 
         for (int j = 1; j <= m; j++)
-            eggFloor[1][j] = j;
+            eggFloor[1][(@IndexFor("eggFloor[1]") int)j] = j;
 
         // Using bottom-up approach in DP
 
         for (int i = 2; i <= n; i++) {
             for (int j = 2; j <= m; j++) {
-                eggFloor[i][j] = Integer.MAX_VALUE;
+                eggFloor[i][(@IndexFor("eggFloor[i]") int)j] = Integer.MAX_VALUE;
                 for (x = 1; x <= j; x++) {
-                    result = 1 + Math.max(eggFloor[i - 1][x - 1], eggFloor[i][j - x]);
+                    /* This Line gives array.access.unsafe.high, expression.unparsable.type.invalid warning because the compiler is unable to statically verify that i-1 is less than length of the array "eggFloor"*/
```

Comment:
This requires more investigation: the checker isn't supposed to issue `expression.unparsable.type.invalid` errors unless the programmer (you, in this case) wrote a type that's non-sensical or that the checker can't handle. In this case, I think that's probably `@IndexFor("eggFloor[i-1]")`, because the analysis doesn't do arithmetic in the arguments to annotations.

Reply:
I have made a comment and printed the error message related to it. Please suggest, what is the issue with the annotation.